lib/rx/util/health_check_authorization.rb

```diff
@@ -0,0 +1,35 @@
+module Rx
+  module Util
+    class HealthCheckAuthorization
+      def initialize(env, authorization)
+        @authorization = authorization
+        @env = env
+      end
+
+      def ok?
+        return custom_authorization(@env, @authorization) if @authorization.is_a?(Proc)
+
+        default_authorization(@authorization)
+      end
+
+      private
+
+      def custom_authorization(env, callable)
+        callable.call(env)
+      end
+
+      def default_authorization(key)
+        req_key = api_key(@env, "authorization_token")
+        req_key == key
+      end
+
+      def api_key(env, name)
+        key = "http_#{name}".upcase

```

Comment:
a few questions here:

1) why use a custom `authorization_token` header instead of the standard `authorization`?

2) given that this value is hard-coded in the `default_authorization` method, what do you think about pulling this up into a constant? e.g.:

```ruby
class HealthCheckAuthorization
  HTTP_HEADER = "HTTP_AUTHORIZATION"
end
```

Reply:
The answers in order:
1. True I agree with going by standard `authorization`, the reason I chose the other one is we use the same header name in our custom middleware in our project to handle this authorization and I just used the same. it is changed now.
2. I agree, done